pkgs/tools/inputmethods/ibus-engines/ibus-rime/default.nix

```diff
@@ -22,15 +24,18 @@ stdenv.mkDerivation rec {
     sha256 = "0gdxg6ia0i31jn3cvh1nrsjga1j31hf8a2zfgg8rzn25chrfr319";
   };
 
-  buildInputs = [ gdk-pixbuf glib ibus libnotify librime rime-data ];
+  buildInputs = [ gdk-pixbuf glib ibus libnotify librime ];
   nativeBuildInputs = [ cmake pkg-config ];
 
-  cmakeFlags = [ "-DRIME_DATA_DIR=${rime-data}/share/rime-data" ];
+  cmakeFlags = [ "-DRIME_DATA_DIR=${placeholder "out"}/share/rime-data" ];
 
-  prePatch = ''
-    substituteInPlace CMakeLists.txt \
-       --replace 'DESTINATION "''${RIME_DATA_DIR}"' \
-                 'DESTINATION "''${CMAKE_INSTALL_DATADIR}/rime-data"'
+  rimeDataDrv = symlinkJoin {
+    name = "ibus-rime-data";
+    paths = rimeDataPkgs;
+  };
+
+  postInstall = ''
+    cp -r "$rimeDataDrv/share/rime-data/." $out/share/rime-data/
```

Comment:
```suggestion
    cp -r "${rimeDataDrv}/share/rime-data/." $out/share/rime-data/
```

Reply:
Changed as suggested.

But I think using `rec` and `${rimeDataDrv}` should have almost the same effect as using environment variable `$rimeDataDrv` in this case.
So is it always better to use `rec` and `${var}` instead of the environment variable `var`?

Reply:
@SuperSandro2000
Why using string interpolation even though it's passed as mkDerivation args? This will makes `.overrideAttrs` harder, when `.override` is not suitable.

Reply:
finalAttrs should be able to solve this. Relying on string interpolation is brittle.